internal/cmd/config/config.go

```diff
@@ -412,12 +412,19 @@ func Parse(d string) (*Config, error) {
 						reflect.TypeOf(t).String())
 				}
 			}
-			if result.Controller.Database.ConnMaxIdleTime != "" {
-				t, err := parseutil.ParseDurationSecond(result.Controller.Database.ConnMaxIdleTime)
-				if err != nil {
-					return nil, err
+			if result.Controller.Database.ConnMaxIdleTime != nil {
+				switch t := result.Controller.Database.ConnMaxIdleTime.(type) {
+				case string:
+					durationString, err := parseutil.ParsePath(t)
```

Comment:
You're swallowing the error here.

Reply:
Here, and in both stanzas above (the one you didn't modify contains a bug, the same one that was copypasta'd here), if the value passed in is not a URL, ParsePath returns the original value and a known error. Look at line 337 for a correct usage of ParsePath.

Reply:
Updated, I think I caught all the correct spots. I added some more tests to explicitly test these raw string cases, although interestingly the tests passed without needing to make a change. I think because when `parseutil.ParsePath` calls `url.Parse`, it does not get an error back from and ends up at the final return of this block:

https://github.com/hashicorp/go-secure-stdlib/blob/bf6d78ef5b727b83b2f8f41b473eb9764b446df4/parseutil/parsepath.go#L30-L455

See: https://go.dev/play/p/JDkw7nm1pVF

Still probably a good idea to be defensive and consistent in the usage of `parseutil.ParsePath`, but difficult to cover in a test.